src/terraform-resource/in/in.go

```diff
@@ -134,6 +132,10 @@ func (r Runner) inWithBackend(req models.InRequest, tmpDir string) (models.InRes
 		if err = r.writeBackendStateToFile(targetEnvName, client); err != nil {
 			return models.InResponse{}, err
 		}
+		if err = r.writeJSONPlanToFile(tfOutput); err != nil {
```

Comment:
The current `in.go` changes have a couple edge cases:

- Currently this outputs the plan statefile as well which contains the "stateful_string" resources. This is internal implementation detail and not something I want to leak to the user.
- The `tfOutput` contains the plan binary and JSON data fields and will get printed to the Concourse UI as metadata, these contain secrets and would be very large blobs in the UI
- Even if this was fixed to download the real statefile, the user could be generating a plan for a new env in which case the statefile won't exist yet so `writeBackendStateToFile` will error

I would start over on this file and try to use `client.GetPlanFromBackend` to download the plan JSON and copy the Local path to the output directory.

Reply:
I see your point and you're completely right. Please have a look at the changes in `in.go`. I don't think it makes much sense to change `writeBackendStateToFile` itself as we would end up in reading and writing files which can be avoided. 
I also introduced a `output_planfile` parameter to separate the functionality from `output_statefile`. Metadata should be clean now as well.